lib/Transforms/Coroutines/CoroSplit.cpp

```diff
@@ -1549,8 +1549,10 @@ static void replacePrepare(CallInst *Prepare, CallGraph &CG) {
       for (auto &Use : Cast->uses()) {
         auto CS = CallSite(Use.getUser());
         if (!CS || !CS.isCallee(&Use)) continue;
-        PrepareUserNode->removeCallEdgeFor(CS);
-        PrepareUserNode->addCalledFunction(CS, FnNode);
+        if (auto *CB = dyn_cast<CallBase>(Use.getUser())) {
+          PrepareUserNode->removeCallEdgeFor(*CB);
+          PrepareUserNode->addCalledFunction(CB, FnNode);
+        }
```

Comment:
Looks like there *is* a `CallBase::isCallee`, so this loop can just be this:

```
for (auto &Use : Cast->uses()) {
  auto CB = dyn_cast<CallBase>(Use.getUser());
  if (!CB || !CB->isCallee(&Use)) continue;
  PrepareUserNode->removeCallEdgeFor(*CB);
  PrepareUserNode->addCalledFunction(CB, FnNode);
}
```

(I made this comment before, but I think it might've gotten lost.)

Reply:
Ah, that did get lost unfortunately.  I didn't spot the `isCallee` in the `CallBase`.  Thanks!